google/cloud/spanner/internal/session_pool.cc

```diff
@@ -300,13 +349,22 @@ StatusOr<SessionHolder> SessionPool::Allocate(bool dissociate_from_pool) {
   return Allocate(std::unique_lock<std::mutex>(mu_), dissociate_from_pool);
 }
 
+StatusOr<SessionHolder> SessionPool::Multiplexed() {
+  std::unique_lock<std::mutex> lk(mu_);
+  if (multiplexed_session_ && !multiplexed_session_->is_bad()) {
```

Comment:
Maybe refactor this expression to a function?

```cc
bool SessionPool::HasValidMultiplexedSession(std::unique_lock<std::mutex> const&) {
  return multiplexed_session_ && !multiplexed_session_->is_bad();
}
```

Reply:
Done, although I'm not sure it is a net win.